test/WebJobs.Script.Tests/Rpc/GenericWorkerProviderTests.cs

```diff
@@ -190,5 +220,68 @@ private IEnumerable<IWorkerProvider> TestReadWorkerProviderFromConfig(string lan
                 }
             }
         }
+
+        private (string workerPath, IEnumerable<IWorkerProvider> providers) TestReadWorkerProviderFromConfig(IEnumerable<TestLanguageWorkerConfig> configs, Mock<ILogger<object>> mockLogger, string language = null)
+        {
+            string rootPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            try
+            {
+                foreach (var config in configs)
+                {
+                    string workerPath = Path.Combine(rootPath, config.Language);
+                    Directory.CreateDirectory(workerPath);
+
+                    File.WriteAllText(Path.Combine(workerPath, ScriptConstants.WorkerConfigFileName), config.Json);
+                }
+
+                var scriptHostConfig = new ScriptHostConfiguration();
+                var scriptSettingsManager = new ScriptSettingsManager();
+                var settings = new List<KeyValuePair<string, string>>
+                {
+                    new KeyValuePair<string, string>("workers:config:path", rootPath)
+                };
+                scriptSettingsManager.SetConfigurationFactory(() => new ConfigurationBuilder()
+                    .AddInMemoryCollection(settings).Build());
+                return (rootPath, GenericWorkerProvider.ReadWorkerProviderFromConfig(scriptHostConfig, mockLogger.Object, scriptSettingsManager, language: language));
+            }
+            finally
+            {
+                if (Directory.Exists(rootPath))
+                {
+                    Directory.Delete(rootPath, true);
+                }
+            }
+        }
+
+        private static TestLanguageWorkerConfig MakeTestConfig(string language)
+        {
+            var description = new WorkerDescription()
+            {
+                DefaultExecutablePath = "foopath",
+                DefaultWorkerPath = $"./src/index.{language}",
+                Language = language,
+                Extension = $".{language}"
+            };
+
+            var arguments = new string[] { };
+
+            JObject config = new JObject();
+            config["Description"] = JObject.FromObject(description);
+            config["Arguments"] = JArray.FromObject(arguments);
+
+            string json = config.ToString();
+            return new TestLanguageWorkerConfig()
```

Comment:
What json config file does this map to? Can you provide an example showing what the json looks like?